### ISSUE TITLE
Application freezes up randomly and various services become unresponsive

### ISSUE THREAD

**Issue by mhatvan** [getferdi/ferdi#2174]
**Bug description:**
I upgraded my setup to two 4k monitors a couple of days ago, and I noticed that sometimes the application freezes up now, and various services become unresponsive. It's strange because I can e.g. still use Google Calendar but e.g. WhatsApp UI doesn't react to any clicks anymore and notifications are outdated. 

Sometimes Ferdi reloads itself randomly, which fixes the issue. Other times, I have to restart the application completely.

**Environment:**

- Operating System: Ubuntu 21.10
- Ferdi Version: 5.6.3-nightly.47
- Server: Ferdi

**Comment by vraravam** [comment]
My strong suspicion is to do with the use of Ferdi with a server. Could you please try this without a server?

**Comment by MartinX3** [comment]
I can confirm that behaviour since updating from 5.6.2 to 5.6.3 often.

**Comment by mfortini** [comment]
Same for me. *Not* using a server

**Comment by lockieluke** [comment]
+1 this happens after my mac sleeping for a long time

**Comment by pasnox** [comment]
Same here, linux, no server.

**Comment by klonos** [comment]
Same here, on a Mac. Certain apps freeze, as in not able to click/type in the window any more, but for some like Slack you can still scroll.

Hope this helps. In the meantime, I think I'll revert back to 5.6.2.

**Comment by klonos** [comment]
...nope, that didn't work. Slack still freezes - other apps like Zulip and MS Teams seem to be unaffected.

**Comment by klonos** [comment]
> I have a feeling that this happens when there's a notification...

Yup, not a feeling - that's what happens actually.

**Comment by adrienne** [comment]
FYI, i am still having this after upgrading to 5.6.4 (but oddly, the 5.6.4 nightly build that i've been using as a stopgap does *not* have the problem).

**Comment by mfortini** [comment]
I confirm @adrienne 's findings: 5.6.4 is still affected, I had to revert to 5.6.2

**Comment by vraravam** [comment]
I have started facing the same issue - running the latest nightly.

**Comment by sverhagen** [comment]
I have had this with 5.6.0 or 5.6.1 or whatever I was on: I regularly had to reload a service.

I now have this with 5.6.4 and it is worse: I now will find Ferdi unresponsive entirely, and have `kill ferdi` and restart it all together.

I wanted to also mention that this is happening for me on Ubuntu 21.10. Fwiw (not much), I tried with GPU acceleration both on and off (same results).

My personal feeling is that this is related to my computer suspending.

**Comment by mfortini** [comment]
Still happening in 5.6.8
Had to revert to 5.6.2, which is the latest one not having this issue.
Whatsapp web seems to be particularly affected.
Is there any way of debugging this?

**Comment by sverhagen** [comment]
> Whatsapp web seems to be particularly affected.

Agreed, but I tried running with WhatsApp disabled for some cycles now, and the behavior is the same.

From `syslog` before I decided to kill and restart, not sure if that's relevant:
```
Jan  5 15:27:48 sander-laptop5 ubuntu-appindicators@ubuntu.com[3127]: Using Brute-force mode for StatusNotifierItem :1.270/org/ayatana/NotificationItem/Ferdi1
```

**Comment by sverhagen** [comment]
>My personal feeling is that this is related to my computer suspending.

This holds still true, but the more-precise trigger might be when the external monitor of my laptop is suspended, and the application is forced to the laptop's internal screen. In other words, I can reproduce this as follows:

1. Have the external monitor connected
2. Run Ferdi on the external monitor
3. Disconnect the external monitor
4. See Ferdi being moved to the internal screen
* Expected: Ferdi works
* Actual: Ferdi frozen

Alternatively:

1. Have the external monitor connected
2. Run Ferdi on the *internal screen*
3. Disconnect the external monitor
* Expected: Ferdi works
* Actual: yep, Ferdi indeed works

**Comment by mfortini** [comment]
Same again with ferdi 5.6.10

Here's an strace -F after freezing https://pastebin.com/8rHLEA25

**Comment by adrienne** [comment]
I'm on Windows 10, myself, and using the local server (no Ferdi account). Here's the weird thing for me:

5.6.2 release: works
5.6.3 release: doesn't work
5.6.4nightly7: works (and is the version i've been running for the last several weeks)
5.6.4 release: doesn't work

I dunno if that information is at all useful, but perhaps it will help if you can look at what changed between 5.6.3 and 5.6.4nightly7, and then changed *back* before 5.6.4 was released?
